src/data_structures/arraylist.md

```diff
@@ -0,0 +1,21 @@
+
+### Arraylist
+
+The Array list in the `java.util` package acts as a normal Array with the exception of being able to easily modify its values. In an `Array` you need to resize the data structure to add/remove an item. `ArrayList` does this for you automatically so you don't need to resize the Array manually. **Elements inside an Array list are Objects** Thus `Type` is a wrapper equivalent of a type. 
```

Comment:
I don't like the word "easily" here. Easily as compared to what? You can modify in place with `.set` just as "easily" as with `[]` so what you are really referring to is that this has the ability to grow when you call add or remove.

The way you phrase it "In an `Array` you need to resize the data structure to add/remove an item" implies that it is possible to resize an array. It is not.

> `ArrayList` does this for you automatically so you don't need to resize the Array manually.

This is accurate, but again, following content in order, they haven't had to do this manually. I doubt that its that intuitive of a concept from this sentence alone.

> **Elements inside an Array list are Objects** Thus `Type` is a wrapper equivalent of a type. 

This isn't a great explanation of generic erasure - and generics should be explained before this - but it also is something that will be out of date eventually once `ArrayList<int>` is available.